### PR TITLE
Remove translator deprecations

### PR DESCRIPTION
**Pull request by core23**
## Subject

I am targeting this branch, because this is BC.

## Changelog

```markdown
### Removed
- Remove translator deprecations
```

### CODE REVIEW

**Review by VincentLanglet — CHANGES_REQUESTED** [review]
This is BC-break. 

https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/Translation/TranslatorInterface.php#L24

The old Interface doesn't implement the new one. So if someone pass the old implements as argument, it won't work anymore.

**Comment by core23** [comment]
Oh you're right. I will update this PR to support both interfaces

File: src/Action/AbstractPostArchiveAction.php [hunk 1]
```diff
@@ -43,8 +44,22 @@ abstract class AbstractPostArchiveAction extends Controller
...
 
-    public function __construct(BlogInterface $blog, PostManagerInterface $postManager, TranslatorInterface $translator)
+    /**
+     * @param TranslatorInterface|LegacyTranslator $translator
+     */
+    public function __construct(BlogInterface $blog, PostManagerInterface $postManager, object $translator)
     {
+        // NEXT_MAJOR: Remove usage of LegacyTranslator
+        if (!$translator instanceof LegacyTranslator && !$translator instanceof TranslatorInterface) {
+            throw new \TypeError(sprintf('Argument 3 passed to %s() must be an instance of %s, %s given.', __METHOD__, TranslatorInterface::class, \is_object($translator) ? \get_class($translator) : \gettype($translator)));
```
Comment (VincentLanglet) [inline src/Action/AbstractPostArchiveAction.php:54]:
```suggestion
            throw new \TypeError(sprintf('Argument 3 passed to %s() must be an instance of %s, %s given.', __METHOD__, TranslatorInterface::class, \get_class($translator)));
```

**Review by VincentLanglet — APPROVED** [review]

**Review by franmomu — APPROVED** [review]

**Comment by VincentLanglet** [comment]
@core23 Thanks

**Comment by OskarStark** [comment]
Are you fine with `minor` label?

**Pull request merged**